### ISSUE TITLE
Bug: "subset_data_dir.sh --per-spk" doesn't honor the requested numutt.

### ISSUE THREAD

**Issue by schaltung** [kaldi-asr/kaldi#3567]
The `egs/wsj/s5/utils/subset_data_dir.sh` script doesn't return the number of utterances `per-spk` requested (despite being available in the original `spk2utt` file).

I noticed the stop-boundary on the awk's for-loop is off by 1.

`egs/wsj/s5/utils/subset_data_dir.sh`
```
@@ -108,7 +108,7 @@ elif $speakers; then
elif $perspk; then
  awk '{ n='$numutt'; printf("%s ",$1);
         skip=1; while(n*(skip+1) <= NF-1) { skip++; }
-         for(x=2; x<=NF && x <= n*skip; x += skip) { printf("%s ", $x); }
+         for(x=2; x<=NF && x <= (n*skip+1); x += skip) { printf("%s ", $x); }
         printf("\n"); }' <$srcdir/spk2utt >$destdir/spk2utt
  utils/spk2utt_to_utt2spk.pl < $destdir/spk2utt > $destdir/utt2spk
else
```
TL;DR: that boundary should be `(n*skip+1)` instead or `n*skip`.

**Comment by danpovey** [comment]
Hm.  Can you  submit a PR for the fix and show an example of how affects
the spk2utt files?
